fix(rest): Download licenseinfo file rest endpoint.

smrutis1 opened the pull request:
* Fixed the issue with download license information file rest endpoint and added the documentation for the same. 

closes: #575 

Signed-off-by: Smruti Sahoo <smruti.sahoo@siemens.com>

nutanv1contr commented on rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java:
```diff
@@ -283,12 +283,13 @@ public void downloadLicenseInfo(@PathVariable("id") String id,
 	final String timestamp = SW360Utils.getCreatedOnTime().replaceAll("\\s", "_").replace(":", "_");
-	final OutputFormatInfo outputFormatInfo = licenseInfoService.getOutputFormatInfoForGeneratorClass(generatorClassName);
+	String[] outputGeneratorClassNameWithVariant = generatorClassNameWithVariant.split("::");
+	final OutputFormatInfo outputFormatInfo = licenseInfoService.getOutputFormatInfoForGeneratorClass(outputGeneratorClassNameWithVariant[0]);
```
Please remove white spaces.

nutanv1contr replied:
Could you rename the field generatorClassName to generatorClassNameWithVariantinside inside licenseInfoService.getLicenseInfoFile(..) method also.

smrutis1 replied:
> Please remove white spaces.

Removed it

smrutis1 replied:
> Could you rename the field generatorClassName to generatorClassNameWithVariant inside licenseInfoService.getLicenseInfoFile(..) method also.

Done

nutanv1contr commented on rest/resource-server/src/docs/asciidoc/projects.adoc:
```diff
@@ -247,3 +247,16 @@ include::{snippets}/should_document_link_releases/curl-request.adoc[]
+===== Example request
+include::{snippets}/should_document_get_download_license_info/curl-request.adoc[]
+
+===== Example request parameter
```
Could you document "Request parameters" before "Example request" as it is done for other rest end point documentation.

smrutis1 replied:
Updated the order.

nutanv1contr commented:
Tested the rest end point. Now License info is returned as response to the rest request. Generate license info project feature is now working from rest end point.

nutanv1contr commented:
Code looks good to me now.

maxhbr commented:
> Code looks good to me now.

I have removed the `needs code review` label


> Tested the rest end point. Now License info is returned as response to the rest request. Generate license info project feature is now working from rest end point.

I have removed the `needs general test` label

mcjaeger commented:
(from discussion on slack) small change discussed to make two parameters instead of a combined one.

smrutis1 commented:
Updated the PR by splitting the combined request parameter from `generatorClassNameWithVariant` to `generatorClassName` and `variant` and updated the rest documentation accordingly.

mcjaeger commented:
test restarted, although, maybe there is a real (not a timing) issue:
```
Running org.eclipse.sw360.cvesearch.entitytranslation.CveSearchDataTranslatorTest
Tests run: 5, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 350.654 sec <<< FAILURE!
testWithApacheData(org.eclipse.sw360.cvesearch.entitytranslation.CveSearchDataTranslatorTest)  Time elapsed: 326.111 sec  <<< ERROR!
com.google.gson.JsonSyntaxException: java.io.EOFException: End of input at line 1782 column 5 path $[16].vulnerable_configuration_cpe_2_2[78]
	at com.google.gson.Gson.fromJson(Gson.java:897)
	at com.google.gson.Gson.fromJson(Gson.java:852)
```

mcjaeger commented:
seems like unit testing fails for cve search fails at three branches now ...

mcjaeger commented:
I think we can remove code review but need a small test? (is this correctly understood?)

nutanv1contr commented:
Code looks good.We can remove the code review.

nutanv1contr commented:
Tested the PR again.Functionality is working as expected.
